src/osp/core/array_view.h

```diff
@@ -28,10 +28,56 @@
 #include <Corrade/Containers/ArrayView.h>
 // IWYU pragma: end_exports
 
+#include <iterator>
+
 namespace osp
 {
 
 using Corrade::Containers::ArrayView;
 using Corrade::Containers::arrayView;
 
+template<typename T>
+struct ArrayView2DWrapper
+{
+    constexpr ArrayView<T> row(std::size_t const index) const noexcept
+    {
+        return view.sliceSize(index * columns, columns);
+    }
+
+    ArrayView<T> view;
+    std::size_t columns;
+};
+
+template<typename T>
+constexpr ArrayView2DWrapper<T> as_2d(ArrayView<T> view, std::size_t columns) noexcept
+{
+    return { .view = view, .columns = columns };
+}
+
+/**
+ * @brief slice_2d_row
+ * @param view
+ * @param index
+ * @param size
+ */
+template<typename T>
+ArrayView<T> slice_2d_row(ArrayView<T> const& view, std::size_t index, std::size_t size) noexcept
+{
+    return view.sliceSize(index, size);
+}
+
+/**
+ * @brief Anger the address sanitizer for invalid array views / spans / similar containers
```

Comment:
Why do you want to make the address sanitizer mad on purpose?

What were you using this for?

Reply:
There's cases where I had an array view that exists, and I wanted to assert that it points to valid memory. There's no uses of it right now, but at the time it was actually quite helpful.

Reply:
Ah interesting.

In windows there's actually a OS api to let you explicitly check that at runtime.

No idea how to go about doing it in linux though.